### PR TITLE
feat: test triggers

### PR DESCRIPTION
**Pull request by dejanzele**
## Pull request description 
add support for test triggers

## Checklist (choose whats happened)

- [ ] breaking change! (describe)
- [x] tested locally
- [ ] tested on cluster
- [x] added new dependencies
- [ ] updated the docs
- [x] added a test

## Breaking changes

-

## Changes

- Added triggers service
- Extracted prepare tests logic from TestKubeAPI into scheduler package
- Add gomock/mockgen
- Add unit tests to trigger service
- Cleanup openapi yaml

## Fixes

-

### CODE REVIEW

**Review by vsukhin — APPROVED** [review]
fix failed checks

File: internal/app/api/v1/testtriggers.go [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+	testtriggersmapper "github.com/kubeshop/testkube/pkg/mapper/testtriggers"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/kubeshop/testkube/pkg/api/v1/testkube"
+	"github.com/kubeshop/testkube/pkg/crd"
+)
+
+// CreateTestTriggerHandler is a handler for creating test trigger objects
+func (s *TestkubeAPI) CreateTestTriggerHandler() fiber.Handler {
```
Comment (exu) [inline internal/app/api/v1/testtriggers.go:21]: Just idea: good place to abstract (or use existing) into some generics based HTTP REST client - too much repeated code - we could discuss @dejanzele @vsukhin about this?

Comment (vsukhin) [reply internal/app/api/v1/testtriggers.go:21]: Yes, we need. it's kinda technical debt

File: pkg/triggers/watcher.go [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+)
+
+func (s *Service) runWatcher(ctx context.Context) {
+	f := informers.NewSharedInformerFactory(s.clientset, 0)
+	podInformer := f.Core().V1().Pods()
+	deploymentInformer := f.Apps().V1().Deployments()
+	// daemonsetInformer := w.f.Apps().V1().DaemonSets()
+	// statefulsetInformer := w.f.Apps().V1().StatefulSets()
+	// serviceInformer := w.f.Core().V1().Services()
+	// ingressInformer := w.f.Networking().V1().Ingresses()
```
Comment (exu) [inline pkg/triggers/watcher.go:23]: do we enable them?

Comment (exu) [reply pkg/triggers/watcher.go:23]: maybe we can configure which one to listen?

File: pkg/triggers/watcher.go [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+	"github.com/kubeshop/testkube-operator/pkg/validation/tests/v1/testtrigger"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/tools/cache"
+)
+
+func (s *Service) runWatcher(ctx context.Context) {
+	f := informers.NewSharedInformerFactory(s.clientset, 0)
+	podInformer := f.Core().V1().Pods()
```
Comment (exu) [inline pkg/triggers/watcher.go:18]: We could pass them as some array with enabled/disabled property.

Comment (exu) [reply pkg/triggers/watcher.go:18]: and then configure which one is handled

File: pkg/triggers/watcher.go (hunk 2 above)
Comment (dejanzele) [reply pkg/triggers/watcher.go]: I enabled all of them

File: pkg/triggers/watcher.go (hunk 3 above)
Comment (dejanzele) [reply pkg/triggers/watcher.go:19]: I enabled all of them, it won't interfere or cause load

**Review by exu — APPROVED** [review]

File: internal/pkg/api/repository/result/interface.go [hunk 4]
```diff
@@ -54,7 +55,7 @@ type Repository interface {
 	// StartExecution updates execution start time
 	StartExecution(ctx context.Context, id string, startTime time.Time) error
 	// EndExecution updates execution end time
-	EndExecution(ctx context.Context, execution testkube.Execution) error
+	EndExecution(ctx context.Context, id string, endTime time.Time, duration time.Duration) error
```
[the remark this answers is not in the document]
Comment (dejanzele) [reply internal/pkg/api/repository/result/interface.go]: nice catch, resolved

**Pull request merged**